packages/server/integrations/TaskIntegrationManagerFactory.ts

```diff
@@ -33,7 +37,8 @@ export interface TaskIntegrationManager {
     assigneeName: string,
     teamName: string,
     teamDashboardUrl: string,
-    issueId: string
+    issueId: string,
+    integrationHash?: string
```

Comment:
`issueId` here is a number like `7` whereas the `integrationHash` contains the `instanceId` that we need. See comment [here](https://github.com/ParabolInc/parabol/pull/7007#discussion_r941267317) for context

Reply:
What if we remove `issueId` here and only be using `integrationHash` for all integrations?

As for example for jira we do `JiraIssueId.split(issueId)`, same we do in `integrationHash.split`.

Also we can try to get some data from provider and store in constructor like we do for JiraServer.

```
    const {serverBaseUrl, consumerKey, consumerSecret} = this.provider
    const {accessToken, accessTokenSecret} = this.auth

    this.serverBaseUrl = serverBaseUrl
```    

This is all needed to avoid having optional params that are needed only for a particular integration.

As it requires additional refactoring we can create a separate issue I guess.

Reply:
Yeah, I think it's definitely worth exploring. The following is confusing to me:
 https://github.com/ParabolInc/parabol/blob/5e889841b96397f3c4f48769947d0f0f323b2a8b/packages/server/integrations/github/GitHubServerManager.ts#L82-L86

In our integrations, `issueId` is what we get from the API, but `integrationHash` uses `GitHubIssueId`. To prevent scope creep, I'd like to do the following:

- Create an issue to explore a way of simplifying how `issueId` and `integrationHash` currently work
- See if we can just use `issueId` or `integrationHash`
- Consider renaming `GitHub/Jira/GitLabIssueId.join` to `GitHubIntegrationHash`


CC @mattkrick

Reply:
in a perfect world, the integration hash would a 1-way opaque nonsensical string of characters used to compare equality. A canonical graphql relay server does this when composing IDs by just base64ing it. 

for this use case specifically, `addCreatedBySomeoneElseComment` requires different fields for different implementations. By using the integration hash, we're kinda cramming multiple variables into parameter. I think a clearer expression would be to have the abstract form have a  `params: Record<string, string>` type. Then, each implementation can be more specific about what's inside

e.g. Azure would have `params: {issueId: number, projectId: string}`

Reply:
Yeah, I prefer an object rather than a concatenated string too as TS should save us from bugs. Are you thinking we should replace `IntegrationHash` throughout the app with an object or just in the integration server manager `createTask`s?

Other places `IntegrationHash` is also used in ImportedTask for example: https://github.com/ParabolInc/parabol/blob/8f583d6ff957df3e41dec5e6ee7d4ac47f8b8875/packages/server/database/types/ImportedTask.ts#L25

Once I get clarity on that, I'll create an issue and change this in a separate PR @mattkrick